static/flow/views/program-editor-panel.js

```diff
@@ -83,6 +83,12 @@ var ProgramEditorPanel = function(options) {
             programSpec.name = this.chooseProgramName();
             programSpec.displayedName = "untitled program";
         }
+        else if(programSpec.name == null || programSpec.name == ""){
+            // if we loaded a program copy stored in dataset metadata,
+            // or if the name value is null or undefined
+            // then it might not have a valid file name
+            programSpec.name = this.createDateTimeName("program_");
```

Comment:
Where is this function defined? Should this be `this.chooseProgramName()` instead?

Reply:
This might be something that crept in that actually is part of one of the other pull requests.  I refactored that function name in the tech-debt branch.  I might have mixed up the name change when checking in this branch.  I'll investigate.

Reply:
Checked in a change to fix this  We need to get the other branches merged in to prevent future issues like this.